presto-spi/src/main/java/com/facebook/presto/spi/block/AbstractInterleavedBlock.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.block;
+
+import io.airlift.slice.Slice;
+
+public abstract class AbstractInterleavedBlock
+        implements Block
+{
+    protected final int columns;
+
+    protected abstract Block getBlock(int blockId);
+
+    public abstract InterleavedBlockEncoding getEncoding();
+
+    protected AbstractInterleavedBlock(int columns)
+    {
+        if (columns == 0) {
+            throw new IllegalArgumentException("Number of blocks in BlockListBlock must not be 0.");
+        }
+        this.columns = columns;
+    }
+
+    int getBlockCount()
+    {
+        return columns;
+    }
+
+    /**
+     * Can only be called after the child class is initialized enough that getBlock will return the right value
+     * @return
+     */
+    protected InterleavedBlockEncoding computeBlockEncoding()
```

Comment:
Make this a static that takes a `Block[]` as an argument, then you can remove the comment and the implicit dependency on the class being sufficiently initialized